### ISSUE TITLE
"Copy" and "Copy&Publish" source locale

### ISSUE THREAD

**Issue by digitall-it** [tractorcow-farm/silverstripe-fluent#521]
Seems like when translating a page to a new locale, it will not try to populate it from the global locale, but from the last locale that this page has been published. That means that if I have all translations from Italian (default language) to French, German and Russian in some Word documents, I have some difficulty trying to replace content which is not in italian.

To better understand, I first have done French. All pages have been published with content in italian, then I've replaced the italian text from my translated French document.

Then I've tried to do German and discovered that when you first publish German pages, their default content is not pulled from the Italian but from the French (the last locale the page has been published to). So I had to continuously switch to Italian to be sure I was copying and pasting the right text in the right positions.

Then I've tried doing russian, and found german text. At this point, I had an Italian to Russian word document on the right, and the CMS with fields filled with german content on the left.

I decided to make myself a nice coffee and write this one.

Somehow related to #402.

**Comment by digitall-it** [comment]
I'm now filling a series of English pages that come from Russian and remembered this issue!

**Comment by Leapfrognz** [comment]
We are also getting this issue. Eg creating a new trsnalation for French, it is pre-opulated from Chinese, which was the last translation added. Are there any work-arounds for this?

**Comment by tractorcow** [comment]
Yes the current 4.x version of fluent has some issue where source locale isn't always obvious to the content author.

5.x will add some new tools that make things clearer. E.g. explicit "copy <X> to <Y>" actions.

In 4.x the source problem is a little hard to solve, since a page may never be written in the "default locale" at all (e.g. creating a page in another locale). The default locale actually has no real significance in this version of fluent, other than it's the locale used for the `/` home route with no locale suffix, so it's not always the correct source for these pages anyway.

**Comment by tractorcow** [comment]
Example tools: ![](https://user-images.githubusercontent.com/936064/68266985-9a896e80-00b5-11ea-9c63-6a06c9a8e410.png)

You can use `dev-develop` if you want to try these out, but keep in mind these are super unstable and may break until we stabilise and merge this into master.

**Comment by t3hn0** [comment]
Since this is still opened issue, I can contribute with my workaround...We had same problems on our project.

I divided this in two parts:
1) writing changes to DB
You have many tables in your database which represent your instances of Page, default Silverstripe created tables (let's call them **base tables**, like: SiteTree, SiteTree_Live, SiteTree_versions, Page, ....) and fluent created **_localised tables**. Localised tables contain translationed fields for every language. Fluent writes changes to localised tables for current locale and also writes current changes to base table. So when you edit page in any locale, values from that locale will be stored in base table.

2) creating translation
When you create translation of existing page in new locale (let's say we want to create en_US) with "Copy to draft" or "Copy & publish" button (this goes for Fluent 4.x and Fluent 5.x), it will retain default values from base table since right join to SiteTree_localised won't exist with Locale=en_US in "JOIN ON" condition.

Conclusion: 
problem 1 was easier to solve, by creating Extension which extends default FluentSiteTreeExtension and change it with injector. Extension prevents overwriting of certain fields in base table if current locale is not IsGlobalDefault.


yml config:
```
SilverStripe\Core\Injector\Injector:
  TractorCow\Fluent\Extension\FluentSiteTreeExtension:
    class: Innovatif\Decorators\MyFluentSiteTreeExtension
```

decorator:
```
<?php
namespace Innovatif\Decorators;

use TractorCow\Fluent\Model\Locale;
use TractorCow\Fluent\Extension\FluentSiteTreeExtension;

class GOVFluentSiteTreeExtension extends FluentSiteTreeExtension
{

    protected function localiseManipulationTable(&$manipulation, $table, $localeTable, $localisedFields, Locale $locale)
    {
        parent::localiseManipulationTable($manipulation, $table, $localeTable, $localisedFields, $locale);

        // if SiteTree exists && current locale !== default locale, unset SiteTree manipulation for certain fields
        if ($this->owner->exists() && ! $locale->IsGlobalDefault) {
            // you can implement custom logic of fields you don't want to override
            $skip_fields = [
                'Title',
                'MenuTitle',
                'URLSegment',
                'Content',
                'MetaDescription',
                'ExtraMeta'
            ];

            if (isset($manipulation['SiteTree'])) {
                foreach ($skip_fields as $field) {
                    if (isset($manipulation['SiteTree']['fields'][$field])) {
                        unset($manipulation['SiteTree']['fields'][$field]);
                    }
                }
            }
            if (isset($manipulation['SiteTree_Live'])) {
                foreach ($skip_fields as $field) {
                    if (isset($manipulation['SiteTree']['fields'][$field])) {
                        unset($manipulation['SiteTree']['fields'][$field]);
                    }
                }
            }
        }
    }
}
```

**Comment by tractorcow** [comment]
I think you could get a similar success if you configure the "Failover" of your locales. Make sure that each locale specifies the default locale as a failover, and you should get the same effect.

E.g. (with nz as default locale)

![image](https://user-images.githubusercontent.com/936064/82405057-fa29f300-9ab6-11ea-998c-350a9d71d94f.png)
